hela/forest.py

```diff
@@ -0,0 +1,251 @@
+import os
+import json
+
+import numpy as np
+from sklearn import metrics, multioutput
+import joblib
+
+from .dataset import load_dataset, load_data_file
+from .models import Model
+from .plot import (plot_predicted_vs_real, plot_feature_importances,
+                   plot_posterior_matrix)
+from .wpercentile import wpercentile
+
+__all__ = ['RandomForest', 'generate_example_data']
+
+
+def train_model(dataset, num_trees, num_jobs, verbose=1):
+    pipeline = Model(num_trees, num_jobs,
+                     names=dataset.names,
+                     ranges=dataset.ranges,
+                     colors=dataset.colors,
+                     verbose=verbose)
+    pipeline.fit(dataset.training_x, dataset.training_y)
+    return pipeline
+
+
+def test_model(model, dataset, output_path):
+    if dataset.testing_x is None:
+        return
+
+    pred = model.predict(dataset.testing_x)
+    r2scores = {name_i: metrics.r2_score(real_i, pred_i)
+                for name_i, real_i, pred_i in
+                zip(dataset.names, dataset.testing_y.T, pred.T)}
+    print("Testing scores:")
+    for name, values in r2scores.items():
+        print("\tR^2 score for {}: {:.3f}".format(name, values))
+
+    fig = plot_predicted_vs_real(dataset.testing_y, pred, dataset.names,
+                                 dataset.ranges)
+    fig.savefig(os.path.join(output_path, "predicted_vs_real.pdf"),
+                bbox_inches='tight')
+    return r2scores
+
+
+def compute_feature_importance(model, dataset, output_path):
+    regr = multioutput.MultiOutputRegressor(model, n_jobs=1)
+    regr.fit(dataset.training_x, dataset.training_y)
+
+    forests = [i.rf for i in regr.estimators_] + [model.rf]
+
+    fig = plot_feature_importances(
+        forests=[i.rf for i in regr.estimators_] + [model.rf],
+        names=dataset.names + ["joint prediction"],
+        colors=dataset.colors + ["C0"])
+
+    fig.savefig(os.path.join(output_path, "feature_importances.pdf"),
+                bbox_inches='tight')
+    return np.array([forest_i.feature_importances_ for forest_i in forests])
+
+
+class RandomForest(object):
+    """
```

Comment:
Understood, I agree this could be improved. I was going for the [scikit-learn](https://scikit-learn.org/stable/modules/linear_model.html#ordinary-least-squares) naming style here, where the name of the trainable model class is the same as the name of the technique it uses. 

Reflecting on the code a bit, it might make sense to change the names: 
* `Model` -> `RandomForestWrapper`, since that's a descriptive name for what the class does
* `RandomForest` -> `Model`

That way the calls will make sense (i.e. "train the model", "predict from the model"):
```python
from hela import Model

m = Model(...)
m.train(...)
m.predict(..)
```
Can you think of a less vague name than Model? @cefisher – what do you think?

Reply:
After looking over the `rfretrieval` file, maybe `Retrieval` would be a better name?

Reply:
`Retrieval` sounds definitely better to me.